src/main/java/org/humancellatlas/ingest/project/ProjectService.java

```diff
@@ -175,4 +183,38 @@ private ProjectBag gather(Project project) {
         return new ProjectBag(projects, envelopes);
     }
 
+    public Page<Project> filterProjects(SearchFilter searchFilter, Pageable pageable) {
+        List<Criteria> criteria_list = new ArrayList<>();
```

Comment:
consider extracting the criteria building part to its own function, to keep abstraction level constant.